src/UserRepository/Htpasswd.php

```diff
@@ -20,17 +21,20 @@
  */
 class Htpasswd implements UserRepositoryInterface
 {
-    use UserTrait;
-
     /**
      * @var string
      */
     private $filename;
 
+    /**
+     * @var UserInterfaceFactory
+     */
+    private $userFactory;
+
     /**
      * @throws Exception\InvalidConfigException
      */
-    public function __construct(string $filename)
+    public function __construct(string $filename, UserInterfaceFactory $userFactory)
```

Comment:
Right here is a good example of why the factory should likely have an interface (e.g., `UserFactoryInterface`. That way, developers do not need to _extend_ the factory class, but can instead provide their own implementation of the interface.

Reply:
I changed it with a `callable`.